### PR TITLE
PR: Include lineterminator in readline

### LINKED ISSUE #77: Include endlines in readline
HDFS3 differs from typical file handling in that it does not include endlines.  This stops convenient use with wrappers like `io.TextIOWrapper`

```python
In [1]: with open('foo', 'wb') as f:
    f.write('Alice 100\nBob 200\nCharlie 300'.encode())
   ...:     

In [2]: f = open('foo', 'rb')

In [3]: f.readline()
Out[3]: 'Alice 100\n'

In [4]: f.readline()
Out[4]: 'Bob 200\n'

In [5]: f.readline()
Out[5]: 'Charlie 300'

In [6]: import hdfs3

In [8]: hdfs = hdfs3.HDFileSystem()

In [9]: with hdfs.open('/tmp/test/text.1.txt', 'wb') as f:
    f.write('Alice 100\nBob 200\nCharlie 300'.encode())
   ...:     

In [10]: f = hdfs.open('/tmp/test/text.1.txt', 'rb')

In [11]: f.readline()
Out[11]: 'Alice 100'

In [12]: f.readline()
Out[12]: 'Bob 200'

In [13]: f.readline()
Out[13]: 'Charlie 300'
```

### PR DESCRIPTION
**Pull request by martindurant**
(also refactored how the lines were read; readline at end of file
now returns empty bytestring)

Fixes #77

### CODE REVIEW

**Comment by mrocklin** [comment]
Travis currently is set to test on every PR and on every push to a branch in the main fork.  

I tend to develop in my own fork, `mrocklin/branch-name`, and then issue a PR from there.  I think you tend to develop in the main fork, `dask/branch-name`, and so you get a travis response both for being on the main fork and for issuing a PR.  It's not that big of a deal either way.

Your travis tests are passing just fine actually.  If you click on the red X it will show you that it's the new CI hook by anaconda-build.  cc @koverholt

File: hdfs3/core.py [hunk 1]
```diff
@@ -620,24 +621,27 @@ def readline(self, chunksize=2**16, lineterminator='\n'):
...
-            if len(buff) == 0:   #EOF
-                remains = self.buffer
-                if remains:
-                    self.buffer = b''
-                    return remains
-                raise EOFError
-            buff = (self.buffer + buff)
-            self.lines = buff.split(lineterminator)
-        return self.lines.pop(0)
+        lines = getattr(self, 'lines', deque([]))
```
Comment (mrocklin) [inline hdfs3/core.py]: Should we just have `self.lines = deque([])` during file initialization?

**Comment by mrocklin** [comment]
+1

**Comment by koverholt** [comment]
Regarding the anaconda.org failures, it looks like this PR was just out of sync with master and was using the old .binstar.yml values. Assuming the linux builders are up, this should work in future PRs.

**Pull request merged**